network/p2p/p2ptest/client.go

```diff
@@ -28,63 +29,100 @@ func NewClient(
 	clientNodeID ids.NodeID,
 	serverNodeID ids.NodeID,
 ) *p2p.Client {
-	clientSender := &enginetest.Sender{}
-	serverSender := &enginetest.Sender{}
+	return NewClientWithPeers(
+		t,
+		ctx,
+		clientNodeID,
+		map[ids.NodeID]p2p.Handler{serverNodeID: handler},
+	)
+}
 
+// NewClientWithPeers generates a client to communicate to a set of peers
+func NewClientWithPeers(
+	t *testing.T,
+	ctx context.Context,
+	clientNodeID ids.NodeID,
+	peers map[ids.NodeID]p2p.Handler,
+) *p2p.Client {
+	clientSender := &enginetest.Sender{}
 	clientNetwork, err := p2p.NewNetwork(logging.NoLog{}, clientSender, prometheus.NewRegistry(), "")
 	require.NoError(t, err)
 
-	serverNetwork, err := p2p.NewNetwork(logging.NoLog{}, serverSender, prometheus.NewRegistry(), "")
-	require.NoError(t, err)
+	peerSenders := make(map[ids.NodeID]*enginetest.Sender)
+	peerNetworks := make(map[ids.NodeID]*p2p.Network)
+	for nodeID := range peers {
+		peerSenders[nodeID] = &enginetest.Sender{}
+		serverNetwork, err := p2p.NewNetwork(logging.NoLog{}, peerSenders[nodeID], prometheus.NewRegistry(), "")
+		require.NoError(t, err)
+		peerNetworks[nodeID] = serverNetwork
+	}
 
 	clientSender.SendAppGossipF = func(ctx context.Context, _ common.SendConfig, gossipBytes []byte) error {
 		// Send the request asynchronously to avoid deadlock when the server
 		// sends the response back to the client
-		go func() {
-			require.NoError(t, serverNetwork.AppGossip(ctx, clientNodeID, gossipBytes))
-		}()
+		for nodeID := range peers {
+			nodeIDCopy := nodeID
+			go func() {
+				require.NoError(t, peerNetworks[nodeIDCopy].AppGossip(ctx, nodeIDCopy, gossipBytes))
+			}()
+		}
 
 		return nil
 	}
 
-	clientSender.SendAppRequestF = func(ctx context.Context, _ set.Set[ids.NodeID], requestID uint32, requestBytes []byte) error {
-		// Send the request asynchronously to avoid deadlock when the server
-		// sends the response back to the client
-		go func() {
-			require.NoError(t, serverNetwork.AppRequest(ctx, clientNodeID, requestID, time.Time{}, requestBytes))
-		}()
+	clientSender.SendAppRequestF = func(ctx context.Context, nodeIDs set.Set[ids.NodeID], requestID uint32, requestBytes []byte) error {
+		for nodeID := range nodeIDs {
+			network, ok := peerNetworks[nodeID]
+			if !ok {
+				return fmt.Errorf("%s is not connected", nodeID)
```

Comment:
We could drop it - and this would give us more similar behavior to how the network sender implementation works. I think one downside is that we are only using this explicitly for testing and I think this makes tests easier to debug.